dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/ContextUtils.java

```diff
@@ -421,4 +423,20 @@ public static String getAttachmentFileName( @Nullable String contentDispositionH
         Matcher matcher = CONTENT_DISPOSITION_ATTACHMENT_FILENAME_PATTERN.matcher( contentDispositionHeaderValue );
         return matcher.matches() ? matcher.group( 1 ) : null;
     }
+
+    /**
+     * Returns the value associated with a double wildcard ({@code **}) in the request mapping.
+     * <p>
+     * As an example, for a request mapping {@code /apps/**} and a request {@code /apps/data-visualizer/index.html},
+     * this method will return {@code data-visualizer/index.html}.
+     *
+     * @param request the {@link HttpServletRequest}.
+     * @return the wildcard value.
+     */
+    public static String getWildcardPathValue( HttpServletRequest request )
```

Comment:
Nice!  Should this also be used by `AppController.js`?

Reply:
Yes we can use this other places too (in Java :-)